### ISSUE TITLE
Indentation gets removed

### ISSUE THREAD

**Issue by james10424** [microsoft/pylance-release#3961]
<h3> Environment data </h3>

-   Language Server version: 2023.2.21
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.10.3
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python
[
    {}
]
```

<h3> Repro Steps </h3>

1. Paste the above snippet
1. Set your editor to "Python" if not already
1. Move cursor in between `{}`
1. Hit enter

<h3> Expected behavior </h3>

The indentation should be:
```
[
    {
        # <--- here
    }
]
```

<h3> Actual behavior </h3>

The indentation is:
```
[
    {
    # <--- here
    }
]
```

And if you hit undo, it will revert back to the expected indentation before the original state. So it looks like something is deleting the indentation.

You can test that with a more nested object, but the indentation will only have 4 spaces. Example:
```
[
    {
        "a": {
    # <--- here
        }
    }
]
```

<h3> Logs </h3>

```
(2674) Background analysis message: setFileOpened
(2674) Background analysis message: markFilesDirty
(2674) [FG] parsing: Untitled-2 (0ms)
(2674) [FG] binding: Untitled-2 (0ms)
(2674) Background analysis message: setFileOpened
(2674) Background analysis message: markFilesDirty
(2674) [FG] parsing: Untitled-2 (1ms)
(2674) [FG] binding: Untitled-2 (0ms)
(2674) Background analysis message: analyze
(2674) [BG(1)] analyzing: Untitled-2 ...
(2674) [BG(1)]   parsing: Untitled-2 (0ms)
(2674) [BG(1)]   binding: Untitled-2 (0ms)
(2674) [BG(1)]   checking: Untitled-2 (4ms)
(2674) [BG(1)] analyzing: Untitled-2 (5ms)
(2674) Background analysis message: resumeAnalysis
(2674) Background analysis message: getSemanticTokens delta
(2674) [BG(1)] getSemanticTokens delta previousResultId:1676496645596 at Untitled-2 (0ms)
```

**Comment by rchiodo** [comment]
Thanks for the issue. I can repro. 

We should follow PEP8 for indentation and it looks like this case isn't working.

**Comment by rchiodo** [comment]
This issue has been partially fixed in prerelease version 2023.4.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023431-19-April-2023-PreRelease)

There's still a problem with tuples though:
https://github.com/microsoft/pylance-release/issues/4259
